org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/DocumentSymbolHandler.java

```diff
@@ -43,6 +46,9 @@
 	private void collectChildren(ITypeRoot unit, IJavaElement[] elements, ArrayList<SymbolInformation> symbols)
 			throws JavaModelException {
 		for(IJavaElement element : elements ){
+			if (element.getElementName() == null || element.getElementName().matches(REGEX)) {

```

Comment:
Checking the element name is unfortunately incorrect. While this is a corner case, we can still miss legit elements
![may-11-2017 16-42-55](https://cloud.githubusercontent.com/assets/148698/25971098/1bb5b99c-3669-11e7-8ac8-7efb2b622adb.gif)